sklearn/multiclass.py

```diff
@@ -559,62 +578,29 @@ def decision_function(self, X):
         """
         check_is_fitted(self, 'estimators_')
 
-        predictions = np.vstack([est.predict(X) for est in self.estimators_]).T
-        confidences = np.vstack([_predict_binary(est, X) for est in self.estimators_]).T
-        return _ovr_decision_function(predictions, confidences,
-                                      len(self.classes_))
+        indices = self.pairwise_indices_
+        if indices is None:
+            Xs = [X] * len(self.estimators_)
+        else:
+            Xs = [X[:, idx] for idx in indices]
```

Comment:
Is this case tested? If not please add a dedicated test in `test_multiclass.py` with `SVC(kernel='precomputed')` and check the expected shape of the output.